source/common/formatter/substitution_formatter.cc

```diff
@@ -536,18 +527,19 @@ class StreamInfoAddressFieldExtractor : public StreamInfoFormatter::FieldExtract
 // Ssl::ConnectionInfo std::string field extractor.
 class StreamInfoSslConnectionInfoFieldExtractor : public StreamInfoFormatter::FieldExtractor {
 public:
-  using FieldExtractor = std::function<std::string(const Ssl::ConnectionInfo& connection_info)>;
+  using FieldExtractor =
+      std::function<absl::optional<std::string>(const Ssl::ConnectionInfo& connection_info)>;
 
   StreamInfoSslConnectionInfoFieldExtractor(FieldExtractor f) : field_extractor_(f) {}
 
-  std::string extract(const StreamInfo::StreamInfo& stream_info) const override {
+  absl::optional<std::string> extract(const StreamInfo::StreamInfo& stream_info) const override {
     if (stream_info.downstreamSslConnection() == nullptr) {
-      return UnspecifiedValueString;
+      return absl::nullopt;
     }
 
     const auto value = field_extractor_(*stream_info.downstreamSslConnection());
-    if (value.empty()) {
-      return UnspecifiedValueString;
+    if (value.has_value() && value.value().empty()) {
```

Comment:
I think `if (value && value->empty())` reads better. Otherwise, maybe a different name for the `value` variable is worthwhile here. Similar comments for other formatters below.